common/persistence/nosql/nosqlplugin/cassandra/workflow.go

```diff
@@ -724,11 +727,12 @@ func (db *cdb) InsertReplicationTask(ctx context.Context, tasks []*nosqlplugin.R
 
 	shardID := shardCondition.ShardID
 	batch := db.session.NewBatch(gocql.LoggedBatch).WithContext(ctx)
+	timeStamp := db.timeSrc.Now()
```

Comment:
My point of view:
It is incorrect. The field belongs to the ReplicationTask, so an SQL plugin could later support it.
I don't think db should have timeSrc field, this should be filled on the PersistenceManager level and pushed as is to the DB.

Reply:
db object was using `time.Now()` in a few places since the beginning. While we were adding unit tests last year I replaced all `time.Now()`s with a `db.timeSrc.Now()` so the generated queries can be validated in the tests.  This PR is following same pattern.
What you suggest is valid I think. The timestamp should be determined by the upper layer PersistentManager and the `db` is purely translating the entity records to DB queries without adding anything else on top. However that requires some refactoring. I don't think is is a source of bug but it's just better isolation of responsibilities. Let's turn that into a github issue and label with `#up-for-grab`.